### PR TITLE
[KED-2753] Turn on/off pretty naming

### PR DESCRIPTION
**Pull request by rashidakanchwala**
## Description

Pretty names is a feature that capitalises each word in Kedro-Viz so that things look "prettier" on the navigation sidebar and pipeline graph
This ticket makes it possible for users to turn off this feature  using the Settings panel so that it reflects the actual name

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -34,6 +34,14 @@ export const flags = {
   },
 };
 
+export const settings = {
+  prettyName: {
+    name: 'Pretty Name',
+    description: 'Prettify the kedro node name',
```
Comment (studioswong) [inline src/config.js:40]: 'Prettify' sounds ambiguous - I would reword this as something along the lines of ' display a more read-able version of the node name'

**Review by studioswong — CHANGES_REQUESTED** [review]
I've tested this feature and it works well, good work!

There are a few tests that needs to be added to supplement this feature:
- tests on the sidebar (the node-list tests) to ensure it displays the pretty and non-pretty name according to the settings. This can be set up by mocking the state of the prettyName to be passed in as a prop in the tests.
- tests for the metadata panel to ensure the name reflects according to the state. The setup will be similar to the setup above.

It looks like there is currently a failing JS test on the CI so worth double checking on this as well.

File: src/components/node-list/node-list.test.js [hunk 2]
```diff
@@ -142,6 +143,64 @@ describe('NodeList', () => {
...
+        ['cat', true],
+        ['dog', true],
+        ['elephant', true],
+        ['giraffe', true],
+        ['horse', true],
+        ['pig', true],
+        ['sheep', true],
+        ['parameters', false],
+        ['params:pipeline100.data_science.plankton', false],
+        ['params:rabbit', false],
```
Comment (studioswong) [inline src/components/node-list/node-list.test.js:176]: This set of tests, while does test the displayed title of the wrapper, is more structured to test the `enabled` feature as set out in the other tests - we won't need to test whether they are disabled but only need to test the names, so you can take out the check for the `disabled` class here. This will also apply to the test below from lines 184 - 199 :)

Comment (rashidakanchwala) [reply src/components/node-list/node-list.test.js:176]: Now, i get it . i was wondering why there's true and false when i copy/pasted it from another tests. thanks I have removed it :D

File: src/components/metadata/metadata.test.js [hunk 3]
```diff
@@ -66,6 +67,40 @@ describe('MetaData', () => {
       // Should show all 20 values
       expect(parametersRow().find('.pipeline-metadata__value').length).toBe(20);
     });
+
+    it('when pretty name is turned off the metadata title displays the full node name', () => {
+      const props = {
+        nodeId: salmonTaskNodeId,
```
Comment (studioswong) [inline src/components/metadata/metadata.test.js:73]: It would be better to set up a node that has a more distinctive difference between their pretty name and non-pretty name other than `salmon` - the `params:pipeline100.data_science.plankton` node would work much better.

**Review by studioswong — APPROVED** [review]
Thanks for adding the tests! I highly recommend switching to use the parameter node other than the salmon node for the metadata tests in testing the pretty name, as that node name includes the difference of capitalisation as well as the underscore, which would be much more effective in testing the pretty name feature.

**Review by limdauto — APPROVED** [review]
LGTM!
